### ISSUE TITLE
sql query with REGEXP_MATCH() fails due to sql not parsed correctly

### ISSUE THREAD

**Issue by taimingl** [openobserve/openobserve#2984]
### Which OpenObserve functionalities are the source of the bug?

log search

### Is this a regression?

Yes

### Description

When using `REGEXP_MATCH()` in a sql query, the query request encounters a `Error while fetching data` error. Below is simple example:

Querying log
```
{
  "_timestamp": 1709165734047787,
  "body": "2024-02-29 00:15:30 15.128.22.213 GET /Administradores_Elina/service-worker.js - 443 - 15.128.69.103 Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/122.0.0.0+Safari/537.36 https://ccwsqa/Administradores_Elina 200 0 0 6 15.159.53.87,+15.128.69.103:54910",
  "dropped_attributes_count": 0,
  "log_file_name": "u_ex240229_x.log",
  "severity": ""
}
```
with query:
````
SELECT REGEXP_MATCH(body, '(?P<timestamp>[^\s]+ [^\s]+) (?P<client_ip>[^\s]+)') 
FROM "default"
````

### Please provide the exception or error you saw

```true
Error log:
thread 'tokio-runtime-worker' panicked at src/service/search/datafusion/rewrite.rs:264:69:
called `Result::unwrap()` on an `Err` value: ParserError("Expected ), found: P at Line: 1, Column 39")
```

### Please provide the version you discovered this bug in (check about page for version information)

```true
Version: V0.9.1
Commit Hash: bcc5f7c8658d4dc2464c6bcad11847214f74b859
Build Date: 3/14/2024
```

**Comment by haohuaijin** [comment]
replace the query to follow can solve this issue, 
```
SELECT REGEXP_MATCH(body, '(?P<timestamp>[^\\s]+ [^\\s]+) (?P<client_ip>[^\\s]+)') 
FROM "default"
```
the error is also appear in datafusion, related to escape characters
<img width="1231" alt="image" src="https://github.com/openobserve/openobserve/assets/47685845/d16fe494-f34b-4e0d-8bc5-b4db13d0a370">

**Comment by taimingl** [comment]
hey, @haohuaijin I tried adding `\` and saw the same error. But I think I found the issue. For below query,
```
SELECT regexp_match(body, '(?P<timestamp>[^\\s]+ [^\\s]+) (?P<client_ip>[^\\s]+)') 
FROM "default" 
```
In `service::search::datafusion::exec::merge_rewrite_sql()` function, before it's passed to `rewrite::remove_where_clause()`, nested double quotation marks, `" "" "` , were added in the SELECT, transferring the query to be 
```
SELECT "regexp_match(tbl.body,Utf8("(?P<timestamp>[^\s]+ [^\s]+) (?P<client_ip>[^\s]+)"))"
FROM "default" 
```
causing `Parser::parse_sql` to escape incorrectly. A quick workaround is by user adding alias to regexp_match() function, such as
```
SELECT regexp_match(body, '(?P<timestamp>[^\s]+ [^\s]+) (?P<client_ip>[^\s]+)') as ret
FROM "default" 
```
which works as expected, even with single `\`. 

I tried fixing by changing either inner or outer `""` to single `''`, both led to different issues. Curious to hear how you might fix this.
